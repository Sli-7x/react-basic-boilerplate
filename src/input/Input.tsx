import * as React from 'react';
import { StyledInput, ErrorText } from './Input.style';

export interface IInputProps {
  onChange: (evenr: Event) => void;
  value: string | number;
  label?: string;
  placeholder?: string;
  inputStyle?: { [key: string]: any };
  maxLength?: number;
  minLength?: number;
  errorText?: any;
  type?: string;
}

export class Input extends React.Component<IInputProps, {}> {
  public static defaultProps = {
    type: 'text',
  };

  constructor(props: IInputProps) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  // React.ChangeEvent<Event>
  handleChange(event: any) {
    this.props.onChange(event);
  }

  getErrors(errors: string | string[]) {
    if (errors) {
      if (Array.isArray(errors) && errors.length > 0) {
        return errors.map((val: string, i: number) => <ErrorText key={i}>{val}</ErrorText>);
      }
      if (typeof errors === 'string') {
        return <ErrorText>{errors}</ErrorText>;
      }
    }
  }

  render() {
    const { value, label, placeholder, inputStyle, maxLength, minLength, type, errorText } = this.props;

    const labelComp = label ? <label>{label}</label> : '';

    return (
      <div>
        {labelComp}
        <StyledInput
          style={inputStyle ? { ...inputStyle } : {}}
          type={type}
          value={value}
          onChange={this.handleChange}
          placeholder={placeholder}
          maxLength={maxLength ? maxLength : 255}
          minLength={minLength ? minLength : 0}
          isInvalid={errorText && errorText.length}
        />
        {this.getErrors(errorText)}
      </div>
    );
  }
}
