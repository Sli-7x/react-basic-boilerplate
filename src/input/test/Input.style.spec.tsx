import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { StyledInput, IStyledInput } from '../Input.style';

describe('Input.style', () => {
  describe('StyledInput', () => {
    const props: IStyledInput = {
      isInvalid: false,
    };

    it('should render correctly', () => {
      const tree = renderer.create(<StyledInput {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should cover when passed isInvalid=true', () => {
      const tree = renderer.create(<StyledInput isInvalid={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
