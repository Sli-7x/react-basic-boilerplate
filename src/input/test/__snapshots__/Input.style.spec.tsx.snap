// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.style StyledInput should cover when passed isInvalid=true 1`] = `
<input
  className="sc-bdVaJa euRPep"
/>
`;

exports[`Input.style StyledInput should render correctly 1`] = `
<input
  className="sc-bdVaJa cmfyDO"
/>
`;
